src/apps/explorer/components/SummaryCardsWidget/VolumeChart/useGetVolumeData.ts

```diff
@@ -0,0 +1,89 @@
+import { useEffect, useState } from 'react'
+import { HistogramData, UTCTimestamp } from 'lightweight-charts'
+import { COW_SDK } from 'const'
+import { useNetworkId } from 'state/network'
+import { Network } from 'types'
+import { VolumePeriod } from './VolumeChartWidget'
+import { VolumeDataResponse } from './VolumeChart'
+
+type RawVolumeItem = {
+  timestamp: string
+  volumeUsd: string
+}
+
+export function useGetVolumeData(volumeTimePeriod = VolumePeriod.DAILY): VolumeDataResponse | undefined {
+  const [volumeData, setVolumeDataJson] = useState<VolumeDataResponse | undefined>()
+  const network = useNetworkId() ?? Network.MAINNET
+
+  useEffect(() => {
+    setVolumeDataJson((prevState) => {
+      return { ...prevState, isLoading: true }
+    })
+
+    let rawData: Promise<RawVolumeItem[]>
+    if (volumeTimePeriod === VolumePeriod.DAILY) {
+      rawData = getLastHoursData(network)
+    } else {
+      rawData = getLastDaysData(volumeTimePeriod, network)
+    }
+
+    rawData.then((data: RawVolumeItem[]) => {
+      const volumeData = buildVolumeData(data, volumeTimePeriod)
+      volumeData.data.sort((a, b) => (a.time < b.time ? -1 : 1))
+      setVolumeDataJson({ ...volumeData, isLoading: false })
+    })
+  }, [network, volumeTimePeriod])
+
+  return volumeData
+}
+
+async function getLastHoursData(network: Network): Promise<RawVolumeItem[]> {
+  const data = await COW_SDK[network]?.cowSubgraphApi.getLastHoursVolume(48)
+
+  return (data?.hourlyTotals as RawVolumeItem[]) || []
+}
+
+async function getLastDaysData(
+  period: VolumePeriod.WEEKLY | VolumePeriod.MONTHLY | VolumePeriod.YEARLY,
+  network: Network,
+): Promise<RawVolumeItem[]> {
+  const days = {
+    [VolumePeriod.WEEKLY]: 7 * 2,
+    [VolumePeriod.MONTHLY]: 30 * 2,
+    [VolumePeriod.YEARLY]: 365 * 2,
```

Comment:
Why * 2?

Reply:
This is to calculate the previous period difference in %.

e.g. for the WEEKLY period, we get 14 data points (1 per day). The user only sees the latest 7 and the other 7 are used to calculate the diff %.